docs/writing-a-connector-guidelines.md

```diff
@@ -0,0 +1,263 @@
+for Conduit connectors provides a basic project setup that includes placeholders
+for the source connector code, destination connector code, connector
+specifications, an example of a configuration, etc.
+
+Also included is a `Makefile` with commonly used targets and GitHub workflows
+for linting the code, running unit tests, automatically merging minor dependabot
+upgrades, etc.
+
+## Research the 3rd party system
+
+Researching the 3rd party system for which a Conduit connector is built helps
+understand the capabilities and limitations of the system, which in turn results
+in a better connector design and better work organization.
+
+Some questions that typically need to be answered:
+
+1. **How is the data organized in the system?**
+
+   The way data is organized in the system will affect how the connector is
+   configured (e.g. what data is a user able to read or write to) and how the
+   hierarchy is mapped to records and collections in Conduit.
+2. **What parameters are needed to connect to the system?**
+
+   We generally recommend using connection strings/URIs, if available. The
+   reason for this is that modifying a connector's parameters is a matter of
+   changing an existing string, and not separate configuration parameters in a
+   connector.
+3. **What APIs or drivers are available?**
+
+   The choice is influenced by a number of factors, such as:
+   - Is there a driver available? If a public API and a driver are available, we
+     recommend using a driver, since it's a higher level of abstraction.
+   - In which language should the connector be written?
+   - If the language of choice is Go, is a pure Go driver available? A CGo
+     driver may make the usage and deployment of a connector more complex.
+   - Does the driver expose all the functionalities needed?
+   - How mature is the driver?
+4. **What authentication methods should be supported?**
+
+   If multiple authentication methods are available, then a decision needs to be
+   made about the methods that should be implemented first. Additionally, it
+   should be understood how expired credentials should be handled. For example,
+   a connector won't be able to handle an expired password, but a token can
+   sometimes be refreshed.
+5. **Can the connector be isolated from other clients of the system?**
+
+   In some cases a connector, as a client using a system, might affect other
+   clients, for example when getting messages from a message broker's queue, the
+   message will be delivered to the connector, while other clients might expect
+   the message.
+6. **Are the source/destination specific features supported?**
+
+   Source and destination connectors may have specific requirements (some of
+   them are outlined in later sections). When researching, attention should be
+   paid if those requirements can be fulfilled.
+
+## Development
+
+## Familiarize with the Connector SDK/connector protocol
```

Comment:
Something doesn't seem right here, I suppose the second header would need to be nested below the first?

```suggestion
## Development

### Familiarize with the Connector SDK/connector protocol
```